### ISSUE TITLE
How can we update `json` cpp library?

### ISSUE THREAD

**Issue by AucaCoyan** [federico-terzi/espanso#2028]
### Description

I'm pretty new to the C++ world. Reading the [json C++ library](https://github.com/nlohmann/json?tab=readme-ov-file#integration)I found the annoyingly amount of package managers and build systems that exist on C++ to mark a dependency.

### Motivation

My intention is focused in having a single point of update for the JSON library. I would like to have a place to update a commit or version, like you do in rust:
```toml Cargo.toml
[dependencies]
json = "3.11.3"
```

Or if we use git submodules, a pinned commit to follow. Does anybody (especially someone versed in C++, not like me) has any idea on a better workflow to manage this json library? Maybe we can apply it to wxWidgets too (currently vendored in a .zip file)

### Alternatives

- Find some way to simplify the dependency management
- Just keep copying and pasting the file as we are doing now. (Not a bad solution after all, it works)

### Have you tried building it? Or would you like it?

- [ ] I'd be willing to contribute this feature to Espanso myself.

**Comment by joh6nn** [comment]
I only took a quick look just now, so please double-check and make sure I didn't miss anything, but the only place I saw that actually being used is in the Win32 native stuff. And it looks like most of that is being used to create the Windows System Tray icon.

If that's correct, if we swapped to using [this crate to create tray icons](https://crates.io/crates/tray-item) then we could

- reduce the number of places we're using non-Rust code
- unify / simplify Tray icon handling across Windows and Mac
- add Tray icon support for Linux

Again, please check and make sure I didn't miss anything 'cause I only took a very quick look

**Comment by AucaCoyan** [comment]
Yeah! This looks very promising. I'll need some time to implement a feature and check if all works, but trying is still progress
